src/java/com/zimbra/oauth/handlers/impl/GoogleContactsImport.java

```diff
@@ -0,0 +1,850 @@
+import static com.zimbra.common.mailbox.ContactConstants.A_pager;
+import static com.zimbra.common.mailbox.ContactConstants.A_phoneticCompany;
+import static com.zimbra.common.mailbox.ContactConstants.A_workCity;
+import static com.zimbra.common.mailbox.ContactConstants.A_workCountry;
+import static com.zimbra.common.mailbox.ContactConstants.A_workEmail1;
+import static com.zimbra.common.mailbox.ContactConstants.A_workFax;
+import static com.zimbra.common.mailbox.ContactConstants.A_workMobile;
+import static com.zimbra.common.mailbox.ContactConstants.A_workPhone;
+import static com.zimbra.common.mailbox.ContactConstants.A_workPostalCode;
+import static com.zimbra.common.mailbox.ContactConstants.A_workState;
+import static com.zimbra.common.mailbox.ContactConstants.A_workStreet;
+import static com.zimbra.common.mailbox.ContactConstants.A_workURL;
+
+import java.io.IOException;
+import java.io.StringReader;
+import java.io.UnsupportedEncodingException;
+import java.net.URISyntaxException;
+import java.text.DateFormat;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Calendar;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Set;
+
+import org.apache.commons.httpclient.methods.GetMethod;
+import org.apache.commons.lang.StringUtils;
+import org.apache.http.client.utils.URIBuilder;
+import org.xml.sax.SAXException;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.zimbra.common.service.ServiceException;
+import com.zimbra.common.util.HtmlBodyTextExtractor;
+import com.zimbra.common.util.ZimbraLog;
+import com.zimbra.cs.account.DataSource;
+import com.zimbra.cs.account.DataSource.DataImport;
+import com.zimbra.cs.account.Provisioning;
+import com.zimbra.cs.mailbox.Contact;
+import com.zimbra.cs.mailbox.Mailbox;
+import com.zimbra.cs.mime.ParsedContact;
+import com.zimbra.cs.service.mail.CreateContact;
+import com.zimbra.cs.service.util.ItemId;
+import com.zimbra.oauth.handlers.impl.GoogleOAuth2Handler.GoogleConstants;
+import com.zimbra.oauth.models.OAuthInfo;
+import com.zimbra.oauth.utilities.Configuration;
+import com.zimbra.oauth.utilities.OAuth2Constants;
+import com.zimbra.oauth.utilities.OAuth2Utilities;
+import com.zimbra.oauth.utilities.OAuthDataSource;
+
+/**
+ * The GoogleContactsImport class.<br>
+ * Used to sync contacts from the Google social service.<br>
+ * Based on the original YahooContactsImport class by @author Greg Solovyev.
+ *
+ * @author Zimbra API Team
+ * @package com.zimbra.oauth.handlers.impl
+ * @copyright Copyright © 2018
+ */
+public class GoogleContactsImport implements DataImport {
+
+    /**
+     * The datasource under import.
+     */
+    private final DataSource mDataSource;
+
+    /**
+     * Configuration wrapper.
+     */
+    private final Configuration config = Configuration.getDefaultConfiguration();
```

Comment:
since, we are moving to use LdapConfiguration class to load configurations, similar approach should be followed here.

Reply:
Will rebase to that branch before merging.